test/test_torch.py

```diff
@@ -4147,6 +4147,70 @@ def test_serialization_offset(self):
         self.assertEqual(i, i_loaded)
         self.assertEqual(j, j_loaded)
 
+    def test_serialization_dill_version(self):
+        try:
+            import dill
+        except:
+            # Cannot run this test if dill is not installed
+            return
+        required_dill_version = (0,3,1)
+        dill_version = tuple(int(num) for num in dill.__version__.split('.'))
+        requirement_is_met = dill_version >= required_dill_version
+        x = torch.randn(5, 5)
+
+        with tempfile.NamedTemporaryFile() as f:
+            # First try saving
+            try:
+                torch.save(x, f, pickle_module=dill)
+            except ValueError:
+                error_was_raised = True
+            else:
+                error_was_raised = False
+
+            # If dill version is supported, no error should occur,
+            # but if dill version is not supported, error should occur
+            self.assertTrue(requirement_is_met ^ error_was_raised)
+
+            # Now try loading
+            f.seek(0)
+            try:
+                x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
+            except ValueError:
+                error_was_raised = True
+            else:
+                error_was_raised = False
+            self.assertTrue(requirement_is_met ^ error_was_raised)
```

Comment:
It's nice to be able to check that the error message is human readable using `self.assertRaisesRegex`. In order to accomplish that, you might have to split the test case into two functions: one for supported dill versions and one for unsupported dill versions.